src/components/tabs/__internal__/tabs-header/tabs-header.component.tsx

```diff
@@ -21,21 +32,50 @@ const TabsHeader = ({
   position = "top",
   role,
   extendedLine,
-  alternateStyling,
   noRightBorder = false,
   isInSidebar = false,
 }: TabHeaderProps) => {
+  const [leftScrollOpacity, setLeftScrollOpacity] = useState(0);
+  const [rightScrollOpacity, setRightScrollOpacity] = useState(1);
+
+  const ref = useRef<HTMLDivElement>(null);
+
+  let isScrollable = false;
+
+  const { current } = ref;
+
+  if (position === "top" && current) {
+    isScrollable = current?.scrollWidth > current?.clientWidth;
+  }
+
+  const handleScroll = (
+    e: React.UIEvent<HTMLDivElement> & { target: HTMLDivElement }
+  ) => {
+    const { scrollLeft } = e.target;
+    const scrollRight = getScrollRight(e.target);
+
+    setLeftScrollOpacity(Math.min(getOpacityRatio(scrollLeft), 1));
+    setRightScrollOpacity(Math.min(getOpacityRatio(scrollRight), 1));
+  };
+
+  const throttledHandleScroll = useThrottle(handleScroll, 50);
+
   return (
     <StyledTabsHeaderWrapper isInSidebar={isInSidebar} position={position}>
       <StyledTabsHeaderList
         align={align}
         position={position}
         role={role}
         extendedLine={extendedLine}
-        alternateStyling={alternateStyling}
         noRightBorder={noRightBorder}
         isInSidebar={isInSidebar}
+        onScroll={throttledHandleScroll}
+        leftScrollOpacity={leftScrollOpacity}
+        rightScrollOpacity={rightScrollOpacity}
+        isScrollable={isScrollable}
+        ref={ref}
       >
+        {position === "top" && <StyledTabsBottomBorder />}
```

Comment:
I'm curious as to why we're now using a `div` by itself to represet the bottom border, instead of just using CSS. I notice that previously it was done by a box-shadow on the StyledTabsHeaderList. The replacement looks similar to me, but the new styles I see are pretty complex including specific gradients, so I have to ask - did you check this with UX? (I guess Chromatic will tell us if anything has changed - but I'm interested to know the reason you felt you had to change this.)

Reply:
1. I had to change the `overflow-x` of `StyledTabsHeaderList` to `auto` to allow horizontal scrolling. 
2. That caused the 3px orange outline of the currently focused TabHeader to be clipped by the element boundaries. 
3. To amend that I added padding=3px and margin=-3px to the container which misaligned the `box-shadow` acting as a border. 
4. So to fix that I decided to create a custom element that looks like a border and can be positioned however I like.